cmd/register/register.go

```diff
@@ -122,36 +115,50 @@ func (r *Register) readHardwareDetails() (*models.DomainMetalHammerRegisterMachi
 
 	// now attach neighbors, this will wait up to 2*tx-intervall
 	// if during this timeout not all required neighbors where found abort and reboot.
-	for _, n := range nics {
-		neighbors, err := r.Network.Neighbors(*n.Name)
+	for _, nic := range nics {
+		neighbors, err := r.Network.Neighbors(*nic.Name)
 		if err != nil {
-			return nil, errors.Wrapf(err, "unable to determine neighbors of interface:%s", *n.Name)
+			return nil, errors.Wrapf(err, "unable to determine neighbors of interface:%s", *nic.Name)
 		}
-		n.Neighbors = neighbors
+		nic.Neighbors = neighbors
 	}
 
 	hw.Nics = nics
+	hw.UUID = r.MachineUUID
 
 	blockInfo, err := ghw.Block()
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to get system block devices")
 	}
-	disks := []*models.ModelsV1MachineBlockDevice{}
 	for _, disk := range blockInfo.Disks {
 		if strings.HasPrefix(disk.Name, storage.DiskPrefixToIgnore) {
 			continue
 		}
+		var parts []*models.ModelsV1MachineDiskPartition
+		containsOS := false // not yet allocated
```

Comment:
containsOS is never set to false if i dont miss something

Reply:
You're right, this is an artifact of the previous implementation. We do not need this field at all. I'll remove this